apps/frontend/src/app/5_pages/ProposalPage/components/ProposalAction/ProposalAction.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { FC, useCallback, useMemo } from 'react';
+
+import { t } from 'i18next';
+
+import { SupportedTokens } from '@sovryn/contracts';
+import { Button, ButtonStyle } from '@sovryn/ui';
+
+import { defaultChainId } from '../../../../../config/chains';
+
+import { useAccount } from '../../../../../hooks/useAccount';
+import { useAssetBalance } from '../../../../../hooks/useAssetBalance';
+import { useLoadContract } from '../../../../../hooks/useLoadContract';
+import { translations } from '../../../../../locales/i18n';
+import {
+  ProposalProps,
+  ProposalState,
+} from '../../../BitocracyPage/BitocracyPage.types';
+import { useProposalStatus } from '../../../BitocracyPage/hooks/useProposalStatus';
+import { ProposalActionType } from './ProposalAction.types';
+import { useHandleProposal } from './hooks/useHandleProposal';
+
+type ProposalActionProps = ProposalProps & {
+  contractName: string | undefined;
+  className?: string;
+};
+
+const pageTranslations = translations.proposalPage;
+
+export const ProposalAction: FC<ProposalActionProps> = ({
+  proposal,
+  contractName,
+  className,
+}) => {
+  const { account } = useAccount();
+  const contract = useLoadContract(
+    `governor${contractName}`,
+    'protocol',
+    defaultChainId,
+  );
+
+  const { balance, loading } = useAssetBalance(SupportedTokens.rbtc);
+  const status = useProposalStatus(proposal);
+
+  const isDisabled = useMemo(
+    () => !account || balance.isZero() || !contract || loading,
+    [account, balance, loading, contract],
+  );
+
+  const isActionVisible = useMemo(
+    () =>
+      contract &&
+      (status === ProposalState.Succeeded || status === ProposalState.Pending),
```

Comment:
```suggestion
      (status === ProposalState.Succeeded || status === ProposalState.Queued),
```
`ProposalState.Pending` - is a state just after proposal is created, it's pending to be voted (only valid for 1 block)